.github/workflows/integration-tests-pr.yml

```
@@ -0,0 +1,147 @@
+# This workflow is triggered on pull requests to run the integration
+# tests on the feature branch. It connects the API to the QA database through
+# a tunnel before running the integration tests.
+# Note: The test might fail if the QA database schema does not match the
+# feature branch schema.
+name: PR - Run Integration Tests
+
+on:
+  pull_request:
+    branches:
+      - main
```

Add a filter to the GH event to avoid running integration tests in changes not related to the API. Here the snipped from other actions:
```suggestion
      - main
         paths-ignore:
           - '**.md'
           - "web-app/**"
           - "functions/**"
           - ".github/workflows/web-*.yml"
```
__Identation might be incorrect in the suggested changes__

---

done ✅